charts/px/templates/hooks/pre-install/preinstall-cleanPreviousJobs.yaml

```diff
@@ -0,0 +1,28 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  namespace: kube-system
+  name: px-hook-preinstall-clean
+  labels:
+    heritage: {{.Release.Service | quote }}
+    release: {{.Release.Name | quote }}
+    chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "-15"
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+{{ if ge .Capabilities.KubeVersion.Minor "8" }}
+  backoffLimit: 0
+{{ else }}
+  activeDeadlineSeconds: 30
+{{ end }}
+  template:
+    spec:
+      serviceAccountName: tiller
+      restartPolicy: Never
+      containers:
+      - name: px-clean
+        image: "lachlanevenson/k8s-kubectl:{{ template "px.kubernetesVersion" . }}"
+        command: ['/bin/sh']
+        args: ['-c','kubectl get jobs -nkube-system | grep px-hook | cut -f1 -d " " | xargs -r kubectl delete -nkube-system jobs']
```

Comment:
do we need to specify kubectl here and change the comand to sh? In other words, will below work?

```
        image: "lachlanevenson/k8s-kubectl:{{ template "px.kubernetesVersion" . }}"
        args: ['get jobs -nkube-system | grep px-hook | cut -f1 -d " " | xargs -r kubectl delete -nkube-system jobs']
```

Reply:
@harsh-px that is the only way it works, otherwise it doesnt recognise the shell pipe `|` symbol when passed as `args` to the ENTRYPOINT of the docker container.